jerry-core/config.h

```diff
@@ -39,8 +39,6 @@
  */
 #ifndef CONFIG_MEM_HEAP_AREA_SIZE
 # define CONFIG_MEM_HEAP_AREA_SIZE (512 * 1024)
-#elif CONFIG_MEM_HEAP_AREA_SIZE > (512 * 1024)
-# error "Currently, maximum 512 kilobytes heap size is supported"
```

Comment:
Removing this may be a bad idea, I think it would be better to add `&& !JERRY_CPOINTER_32_BIT` to the condition instead.